src/librustc/ty/print/pretty.rs

```diff
@@ -969,69 +987,131 @@ pub trait PrettyPrinter<'tcx>:
                     _ => p!(write("{}{}", sign_extend(data, size) as i128, i_str))
                 }
             },
-            (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Char) =>
+            // Char
+            (Scalar::Raw { data, .. }, ty::Char) =>
                 p!(write("{:?}", ::std::char::from_u32(data as u32).unwrap())),
```

Comment:
Please don't use `unwrap` here, this is like `bool`!

Reply:
This now prints the integral char value postfixed by `_char` in case of invalid chars.